java-client/src/main/java/com/wavefront/integrations/Wavefront.java

```diff
@@ -142,6 +143,21 @@ private void internalSend(String name, double value, @Nullable Long timestamp, S
         // already connected.
       }
     }
+    try {
+      internalSend0(name, value, timestamp, source, pointTags);
+    } catch (SocketException ex) { // retry at least once on SocketExceptions
+      this.socket = null;
+      try {
+        connect();
+      } catch (IllegalStateException e) {
+        // already connected.
+      }
+      internalSend0(name, value, timestamp, source, pointTags);
+    }
+  }
+
+  private void internalSend0(String name, double value, @Nullable Long timestamp, String source,
```

Comment:
better names please. maybe "internalSendWithOneRetry" and "internalSend", or come up with something better. internalSend0 isn't too helpful